tests/unit/test_packets.py

```diff
@@ -204,3 +206,36 @@ def test__extract_bits(start, nbits):
     data = int(s, 2).to_bytes(2, byteorder="big")
 
     assert packets._extract_bits(data, start, nbits) == int(s[start:start + nbits], 2)
+
+
+def test_ccsds_generator(jpss_test_data_dir):
+    """Test ccsds_generator"""
+    test_data_file = jpss_test_data_dir / "J01_G011_LZ_2021-04-09T00-00-00Z_V01.DAT1"
+    test_packet = packets.create_ccsds_packet()  # defaults
+
+    # From file
+    with test_data_file.open('rb') as f:
+        assert next(packets.ccsds_generator(f))
+
+    # From socket
+    send, recv = socket.socketpair()
+    send.send(test_packet)
+    assert next(packets.ccsds_generator(recv))
+    send.close()
+    recv.close()
+
+    # From bytes
+    # This covers show_progress conditional code and also the end of the iterator
```

Comment:
I'm curious whether we actually want the `show_progress` flag? Could we point users to `tqdm` or another progressbar package instead?

Reply:
Sure! Let's investigate this in #162